tests/ops/service/connectors/test_connector_registry_service.py

```diff
@@ -20,3 +53,397 @@ def test_get_connector_template(self):
+        secondary_mailchimp_dataset,
+    ) = secondary_mailchimp_instance
+    secondary_mailchimp_saas_config = secondary_mailchimp_config.saas_config
+    secondary_mailchimp_dataset.dataset["description"] = mailchimp_template_dataset[
+        "description"
+    ]
+    assert secondary_mailchimp_saas_config["version"] == mailchimp_version
+    assert (
+        secondary_mailchimp_saas_config["description"]
+        == mailchimp_template_config["description"]
+    )
+
+    (
+        tertiary_mailchimp_config,
+        tertiary_mailchimp_dataset,
+    ) = tertiary_mailchimp_instance
+    tertiary_mailchimp_saas_config = tertiary_mailchimp_config.saas_config
+    tertiary_mailchimp_dataset.dataset["description"] = mailchimp_template_dataset[
+        "description"
+    ]
+    tertiary_mailchimp_saas_config = (
+        tertiary_mailchimp_dataset.connection_config.saas_config
+    )
+    assert tertiary_mailchimp_saas_config["version"] == mailchimp_version
+    assert (
+        tertiary_mailchimp_saas_config["description"]
+        == mailchimp_template_config["description"]
+    )
+
+    (
+        secondary_sendgrid_config,
+        secondary_sendgrid_dataset,
+    ) = secondary_sendgrid_instance
+    secondary_sendgrid_saas_config = secondary_sendgrid_config.saas_config
+    secondary_sendgrid_dataset.dataset["description"] = sendgrid_template_dataset[
+        "description"
+    ]
+    assert secondary_sendgrid_saas_config["version"] == sendgrid_version
+    assert (
+        secondary_sendgrid_saas_config["description"]
+        == sendgrid_template_config["description"]
+    )
+
+    # mock methods within template instantiation workflow
+    # to produce an updated saas config template
+    load_config_mock.side_effect = load_config_mock_override
+    load_config_with_replacement_mock.side_effect = (
+        load_config_with_replacement_mock_override
+    )
+    load_dataset_with_replacement_mock.side_effect = (
+        load_dataset_with_replacement_mock_override
+    )
+
+    # run update "script"
+    update_saas_configs(registry, db)
+
+    # confirm updates applied successfully
+    secondary_mailchimp_dataset: DatasetConfig = DatasetConfig.filter(
+        db=db,
+        conditions=DatasetConfig.fides_key == secondary_mailchimp_dataset.fides_key,
+    ).first()
+    validation_function(
+        secondary_mailchimp_dataset,
+        mailchimp_template_config,
+        mailchimp_template_dataset,
+        secondary_mailchimp_config.key,
+        secondary_mailchimp_dataset.fides_key,
+    )
+
+    tertiary_mailchimp_dataset: DatasetConfig = DatasetConfig.filter(
+        db=db,
+        conditions=DatasetConfig.fides_key == tertiary_mailchimp_dataset.fides_key,
+    ).first()
+    validation_function(
+        tertiary_mailchimp_dataset,
+        mailchimp_template_config,
+        mailchimp_template_dataset,
+        tertiary_mailchimp_config.key,
+        tertiary_mailchimp_dataset.fides_key,
+    )
```

Comment:
Your tests are clever but it takes awhile to trace through and figure out the comparisons you're making and believe that you're testing all the important things and not just mocking too much functionality away.  

This is just my opinion, I know others disagree here, but I prefer tests to be extremely clear about what they are testing. My tests tend to be less DRY than my application code but I view them as a secondary way to document the code: new users to the codebase can look at a test and quickly see what a function or API endpoint requires and what the expected results are without having to additionally understand unique quirks about the test itself.  I also view tests as the most important way to protect the critical features of your own code, and when unit tests are too clever, I think it's easy for a dev to just delete a unit test rather than take the time to understand it.  I also want to make sure we are testing as much end-to-end as possible too.

Not requesting a rewrite here, you've put a lot of work into this, just providing my viewpoint on tests really